pkg/threescale/backend/backend.go

```diff
@@ -269,6 +484,48 @@ func (a *Application) updateUnlimitedCounter(metric string, incrementBy int) {
 	a.UnlimitedCounter[metric] = current + incrementBy
 }
 
+// adjustLocalState assumes that we have a new remote state (set on a) fetched from 3scale and modifies local state based
+// on the state obtained during cache flushing
+func (a *Application) adjustLocalState(flushingState cachedApp, remoteState LimitCounter) *Application {
+	a.RemoteState = remoteState
+
+	for metric := range a.LimitCounter {
+		lowestPeriod := a.LimitCounter.getLowestKnownPeriodForMetric(metric)
+
+		if lowestPeriod == nil {
+			continue
+		}
+
+		snapped, ok := flushingState.snapshot.LimitCounter[metric][*lowestPeriod]
+		if !ok {
+			continue
+		}
+
+		latest, ok := a.RemoteState[metric][*lowestPeriod]
+		if !ok {
+			continue
+		}
+
+		current := a.LimitCounter[metric][*lowestPeriod].current
```

Comment:
Maybe we can just get `a.LimitCounter[metric][*lowestPeriod]` into a binding and access `.current` and assign it as well as `max` below?